test/testsymboldatabase.cpp

```diff
@@ -2378,6 +2379,19 @@ class TestSymbolDatabase : public TestFixture {
         ASSERT(call && call->function() == f->function());
     }
 
+    void namespacedFunctionInsideExternBlock() {
+        GET_SYMBOL_DB("namespace N {\n"
+                      "    void f();\n"
+                      "}\n"
+                      "extern \"C\" {\n"
+                      "    void f() {\n"
+                      "        N::f();\n"
+                      "    }\n"
+                      "}\n");
+
+        ASSERT(db && errout.str().empty());
```

Comment:
This test case looks really pointless. It did crash without your changes right? You could add a comment `// Do not crash` .